tests/FSharp.Compiler.ComponentTests/FSharpChecker/FindReferences.fs

```diff
@@ -96,3 +96,67 @@ secondA.DoNothing(secondB)
 
         }
 
+[<Fact>]
+let ``Finding references in project`` () =
+    let size = 20
+
+    let project =
+        { SyntheticProject.Create() with
+            SourceFiles = [
+                sourceFile $"File%03d{0}" [] |> addSignatureFile
+                for i in 1..size do
+                    sourceFile $"File%03d{i}" [$"File%03d{i-1}"]
+            ]
+        }
+        |> updateFile "File005" (addDependency "File000")
+        |> updateFile "File010" (addDependency "File000")
+
+    let checker = FSharpChecker.Create(enableBackgroundItemKeyStoreAndSemanticClassification = true)
+
+    project.WorkflowWith checker {
+        findAllReferencesToModuleFromFile "File000" true (expectNumberOfResults 5)
+    }
+
+[<Fact>]
+let ``We find back-ticked identifiers`` () =
+    SyntheticProject.Create(
+        { sourceFile "First" [] with ExtraSource = "let ``foo bar`` x = x + 5" },
+        { sourceFile "Second" [] with ExtraSource = "let foo x = ModuleFirst.``foo bar`` x" })
+        .Workflow {
+            placeCursor "Second" 6 35 "let foo x = ModuleFirst.``foo bar`` x" ["``foo bar``"]
+            findAllReferences (expectToFind [
+                "FileFirst.fs", 6, 4, 15
+                "FileSecond.fs", 6, 12, 35
+            ])
+        }
+
+[<Fact>]
+let ``We find operators`` () =
```

Comment:
Random shot in the dark - did you check if this flow anyhow affects active patterns?
I mean, again, it cannot make things worse there so it's rather a small hope for some accidental improvement:

https://user-images.githubusercontent.com/5451366/203749893-7e7e74f8-352f-4805-bc93-85e948243461.mp4